src/test/java/com/microsoft/sqlserver/jdbc/resiliency/BasicConnectionTest.java

```diff
@@ -0,0 +1,279 @@
+    @Tag(Constants.xAzureSQLDW)
+    public void testCatalog() throws SQLException {
+        String expectedDatabaseName = null;
+        String actualDatabaseName = null;
+        try (Connection c = ResiliencyUtils.getConnection(connectionString); Statement s = c.createStatement()) {
+            expectedDatabaseName = RandomUtil.getIdentifier("resDB");
+            TestUtils.dropDatabaseIfExists(expectedDatabaseName, connectionString);
+            s.execute("CREATE DATABASE [" + expectedDatabaseName + "]");
+            try {
+                c.setCatalog(expectedDatabaseName);
+            } catch (SQLException e) {
+                return;
+            }
+            ResiliencyUtils.killConnection(c, connectionString);
+            try (ResultSet rs = s.executeQuery("SELECT db_name();")) {
+                while (rs.next()) {
+                    actualDatabaseName = rs.getString(1);
+                }
+            }
+            // Check if the driver reconnected to the expected database.
+            assertEquals(expectedDatabaseName, actualDatabaseName);
+        } finally {
+            TestUtils.dropDatabaseIfExists(expectedDatabaseName, connectionString);
+        }
+    }
+
+    @Test
+    @Tag(Constants.xAzureSQLDB) // Switching databases is not supported against Azure, skip/
+    @Tag(Constants.xAzureSQLDW)
+    public void testUseDb() throws SQLException {
+        String expectedDatabaseName = null;
+        String actualDatabaseName = null;
+        try (Connection c = ResiliencyUtils.getConnection(connectionString); Statement s = c.createStatement()) {
+            expectedDatabaseName = RandomUtil.getIdentifier("resDB");
+            TestUtils.dropDatabaseIfExists(expectedDatabaseName, connectionString);
+            s.execute("CREATE DATABASE [" + expectedDatabaseName + "]");
+            try {
+                s.execute("USE [" + expectedDatabaseName + "]");
+            } catch (SQLException e) {
+                return;
+            }
+            ResiliencyUtils.killConnection(c, connectionString);
+            try (ResultSet rs = s.executeQuery("SELECT db_name();")) {
+                while (rs.next()) {
+                    actualDatabaseName = rs.getString(1);
+                }
+            }
+            // Check if the driver reconnected to the expected database.
+            assertEquals(expectedDatabaseName, actualDatabaseName);
+        } finally {
+            TestUtils.dropDatabaseIfExists(expectedDatabaseName, connectionString);
+        }
+    }
+
+    @Test
+    public void testSetLanguage() throws SQLException {
+        String expectedLanguage = "Italiano";
+        String actualLanguage = "";
+        try (Connection c = ResiliencyUtils.getConnection(connectionString); Statement s = c.createStatement()) {
+            s.execute("SET LANGUAGE " + expectedLanguage);
+            ResiliencyUtils.killConnection(c, connectionString);
+            ResiliencyUtils.isRecoveryAliveAndConnDead(c);

```

Comment:
I have kept it separate as not every test requires it. It is mostly used for the basic connection tests, but not the property tests for instance.

Reply:
this is also a boolean but we are never check return. If this is a void then should not be named isBlah. I think we should just put this in kill as discussed since potentially all tests could have this issue. Most of the time it wouldn't be doing anything.